Ihor/Netflix/src/HomePage/MoviePageMain/MoviePageMain.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import Header from "../header";
+import { RouteComponentProps } from "react-router-dom";
+import Movies from "../../Shared/movie";
+import SortPanel from "../sortPanel/sortPanel";
+import Footer from "../../Shared/footer";
+import SearchPanel from "../SearchPanel";
+import "../../index.css";
+import IMovie from "../../Interfaces/IMovie";
+import { parse, stringify } from "query-string";
+import { MoviesRequested } from "../../redux/Actions/FetchActions";
+import { connect, ConnectedProps } from "react-redux";
+
+interface IMoviePageProps {
+  movies: IMovie[];
+}
+
+type MovieDetailsMainProps = IMoviePageProps &
+  RouteComponentProps &
+  PropsFromRedux;
+class MoviePageMain extends React.Component<MovieDetailsMainProps> {
+  handleSearchChange = ({
+    search,
+    searchBy,
+  }: {
+    search: string;
+    searchBy: string;
+  }) => {
+    const queryUrl = parse(this.props.location.search) as {
+      sortBy: string;
+      search: string;
+      searchBy: string;
+    };
+    const { sortBy } = queryUrl;
+    const query = stringify({ ...queryUrl, search, searchBy });
+    this.props.MoviesRequested(sortBy, searchBy, search);
+    this.props.history.push({
+      pathname: "/",
+      search: query,
+    });
+  };
+
+  handleSortChange = (sortBy: string) => {
+    const queryUrl = parse(this.props.location.search) as {
+      searchBy: string;
+      search: string;
+    };
+    const { searchBy, search } = queryUrl;
+    const query = stringify({ ...queryUrl, sortBy });
+    this.props.MoviesRequested(sortBy, searchBy, search);
+    this.props.history.push({
+      pathname: "/",
+      search: query,
+    });
+  };
+
+  componentDidMount() {
+    const queryUrl = parse(this.props.location.search) as {
+      searchBy: string;
+      search: string;
+      sortBy: string;
+    };
+    const { sortBy, searchBy, search } = queryUrl;
+    this.props.MoviesRequested(sortBy, searchBy, search);
+  }
+
+  componentDidUpdate = (prevProps: RouteComponentProps) => {
+    if (this.props.location !== prevProps.location) {
```

Comment:
As I can see, you perform fetch several times. When you handleSortChange and then after history push when componentDidUpdate got triggered. You should add additional check on history action type. It shouldn't be equal to "PUSH"